### DISCUSSION TITLE
Random song sound plays multiple times in the same frame

### DISCUSSION THREAD

**Discussion by TheBrunoRM** [ppy/osu#26491] [Q&A] [answered]
### Type

Performance

### Bug description

When pressing the random button (or F2) in the song select menu, if pressed rapidly, the sound sometimes plays multiple times in the same frame, causing the sound to be louder.

### Screenshots or videos

https://github.com/ppy/osu/assets/38665405/19ed713f-6d84-4e6d-a1ef-3222e9278d6e

### Version

2023.1231.0

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/13925085/compressed-logs.zip)

**Comment by frenzibyte** [comment]
Moving to discussion because I don't see anything solvable here.

**Reply by bdach** [reply]
what do you _mean_ there's "nothing solvable here"?

**Reply by frenzibyte** [reply]
Sample channels queuing up because game got too slow, no? Can this be solved?

**Reply by peppy** [reply]
I don't get it either. This is a serious issue and should be solved. As a developer if you see / hear something horrible and then say "it can't be solved", you probably need a change of perspective.

**Reply by frenzibyte** [reply]
My apologies, then. It seems I underestimated this.

**Reply by frenzibyte** [reply, marked as answer]
https://github.com/ppy/osu/issues/26497
